d2core/d2ui/sprite.go

```diff
@@ -70,7 +74,7 @@ func (s *Sprite) RenderSegmented(target d2interface.Surface, segmentsX, segments
 		for x := 0; x < segmentsX; x++ {
 			idx := x + y*segmentsX + frameOffset*segmentsX*segmentsY
 			if err := s.animation.SetCurrentFrame(idx); err != nil {
-				log.Printf("SetCurrentFrame error %e", err)
+				s.logger.Error("SetCurrentFrame error" + err.Error())
```

Comment:
For anything that shares a logger instance with some type of manager, we need to alter the log messages to be more descriptive. Currently, this message would print as:
```
[UI Manager][ERROR] SetCurrentFrame error ...
```
and that's not at all as useful as
```
[UI Manager][ERROR] Sprite.RenderSegmented: SetCurrentFrame error ...
```